_posts/2023-03-29-langchain-slackbot.md

```diff
@@ -0,0 +1,78 @@
+
+LangChain and LlamaIndex facilitate in-context learning, an emerging paradigm that allows developers to insert context into input prompts, leveraging LLM's reasoning capabilities for generating more relevant and accurate responses. This differs from finetuning, which requires retraining models using custom datasets, often demanding significant computational resources and time.
+
+By addressing data ingestion and indexing, LangChain and LlamaIndex provide a streamlined framework for integrating custom data into LLMs. Their flexible design simplifies incorporating external data sources, enabling developers to focus on creating powerful applications that harness LLMs' full potential.
+
+These tools bridge the gap between external data and LLMs, ensuring seamless integration while maintaining performance. By utilizing LangChain and LlamaIndex, developers can unlock LLMs' true potential and build cutting-edge applications tailored to specific use cases and datasets.
+
+🛣️ The project we built uses both `langchain` and `llama_index` as well as some extra code for the Slack bot itself. The rest of this blogpost covers the technical details of what we implemented. If you want to get your hands dirty and try out a simpler version, feel free to check out [our Generative Chat example](https://github.com/zenml-io/zenml/tree/develop/examples/generative_chat) that was released last week.
+
+## ZenML 🤝 LLM frameworks
+
+There are various terms being tried out to describe this new paradigm — from LLMOps to Big Model Ops. Not only the words used to describe how engineering will work are new, but the underlying structures and frameworks are also being developed from the ground up. We wanted to witness these changes first hand by participating and getting our hands dirty.
+
+In particular, we wanted to experience how users of ZenML might go about using our framework to integrate with these tools and models. We had seen lots of demos showcasing useful and striking use cases, but none were some of complexities around deploying these in production were raised.
+
+Starting out, we wanted to get a feel for the two well-known options for developing on top of large language models — that is, LangChain and Llama Index — by trying out some of their core functionality. In particular, loading a series of documents, doing some kind of querying of those documents and packaging this all up into some kind of frontend user experience seemed like good places to start.
+
+## Implementing the SlackBot
```

Comment:
Maybe break this down more. There are two parts here, one is the pipeline that gets triggered each time we release, and the other is deploying the slackbot that then uses the post-execution workflow to fetch the latest index. I think having two sections would allow you to highlight the challenges in each

Reply:
I'd also make the pipeline super clear i.e. what steps were in there, thats the cool bit